tests/func/test_datachain.py

```diff
@@ -25,34 +33,157 @@ def test_catalog_anon(tmp_dir, catalog, anon):
 
 def test_from_storage(cloud_test_catalog):
     ctc = cloud_test_catalog
-    dc = DataChain.from_storage(ctc.src_uri, catalog=ctc.catalog)
+    dc = DataChain.from_storage(ctc.src_uri, client_config=ctc.catalog.client_config)
     assert dc.count() == 7
 
 
-def test_from_storage_reindex(tmp_dir, catalog):
+def test_from_storage_non_recursive(cloud_test_catalog):
+    ctc = cloud_test_catalog
+    dc = DataChain.from_storage(
+        f"{ctc.src_uri}/dogs", client_config=ctc.catalog.client_config, recursive=False
+    )
+    assert dc.count() == 3
+
+
+def test_from_storage_glob(cloud_test_catalog):
+    ctc = cloud_test_catalog
+    dc = DataChain.from_storage(
+        f"{ctc.src_uri}/dogs*", client_config=ctc.catalog.client_config
+    )
+    assert dc.count() == 4
+
+
+def test_from_storage_as_image(cloud_test_catalog):
+    ctc = cloud_test_catalog
+    dc = DataChain.from_storage(
+        ctc.src_uri, client_config=ctc.catalog.client_config, type="image"
+    )
+    for im in dc.collect("file"):
+        assert isinstance(im, ImageFile)
+
+
+def test_from_storage_reindex(tmp_dir, test_session):
+    tmp_dir = tmp_dir / "parquets"
     path = tmp_dir.as_uri()
+    os.mkdir(tmp_dir)
 
     pd.DataFrame({"name": ["Alice", "Bob"]}).to_parquet(tmp_dir / "test1.parquet")
-    assert DataChain.from_storage(path, catalog=catalog).count() == 1
+    assert DataChain.from_storage(path, session=test_session).count() == 1
 
     pd.DataFrame({"name": ["Charlie", "David"]}).to_parquet(tmp_dir / "test2.parquet")
-    assert DataChain.from_storage(path, catalog=catalog).count() == 1
-    assert DataChain.from_storage(path, catalog=catalog, update=True).count() == 2
+    assert DataChain.from_storage(path, session=test_session).count() == 1
+    assert DataChain.from_storage(path, session=test_session, update=True).count() == 2
+
+
+@pytest.mark.parametrize(
```

Comment:
Could you please explain why these settings are needed? :)

Reply:
Removed them, please ignore. Related to https://github.com/iterative/datachain/pull/294#discussion_r1730478181